File: Backend/src/main/java/com/kas/online_book_shop/service/AuthorService.java
package com.kas.online_book_shop.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.kas.online_book_shop.model.Author;

public interface AuthorService {
    List<Author> getAllAuthors();

    Author saveAuthor(Author author);

    Author updateAuthor(Author author);

    void deleteAuthor(Long id);

    Author getAuthorById(Long id);
    
    Page<Author> getAllAuthor(Pageable pageable);
}

File: Frontend/re-admin/src/service/AuthorService.js
import axios from "axios";

const API_URL = "http://localhost:8081/api/v1/author";

const getAllAuthors = () => {
    return axios.get(API_URL);
}

const deleteAuthor = (authorId) => {
    return axios.delete(API_URL + '/' + authorId);
}

const updateAuthor = (data) => {
    return axios.put(API_URL, data);
}

const getAuthorById = (authorId) => {
    return axios.get(API_URL + '/' + authorId);
}

const addAuthor = (data) => {
    return axios.post(API_URL, data);
}

export {getAllAuthors, deleteAuthor, updateAuthor, getAuthorById, addAuthor};